src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.MoveTypeEditor.cs

```diff
@@ -123,15 +124,33 @@ private async Task<Document> AddNewDocumentWithSingleTypeDeclarationAsync(Docume
             AddPartialModifiersToTypeChain(
                 documentEditor, removeAttributesAndComments: true, removeTypeInheritance: true, removePrimaryConstructor: true);
 
+            // Keep track of any associated directives on any of the nodes we're removing. If those directives are then
+            // contained in the leading trivia of the type we're moving, we'll remove them from there as well.
+            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+            using var _ = PooledHashSet<SyntaxNode>.GetInstance(out var correspondingDirectives);
+
             // remove things that are not being moved, from the forked document.
             var membersToRemove = GetMembersToRemove(root);
             foreach (var member in membersToRemove)
+            {
+                AddCorrespondingDirectives(syntaxFacts, member, correspondingDirectives);
                 documentEditor.RemoveNode(member, SyntaxRemoveOptions.KeepNoTrivia);
+            }
 
             // Remove attributes from the root node as well, since those will apply as AttributeTarget.Assembly and
             // don't need to be specified multiple times
             documentEditor.RemoveAllAttributes(root);
 
+            // Now remove any leading directives on the type-node that actually correspond to prior nodes we removed.
+            var leadingTrivia = State.TypeNode.GetLeadingTrivia().ToSet();
+            foreach (var directive in correspondingDirectives)
+            {
+                if (leadingTrivia.Contains(directive.ParentTrivia))
+                    documentEditor.RemoveNode(directive);
+            }
+
+            RemoveLeadingBlankLinesFromMovedType(documentEditor);
```

Comment:
We want ot call this *after* we remove all the appropriate directives from the node so that the spacing is correct after that happens.